pkg/controller/tf/controller.go

```diff
@@ -220,15 +220,19 @@ func (r *Reconciler) sync(ctx context.Context, krmResource *krmtotf.Resource) (r
 			r.logger.Info("deletion policy set to abandon; abandoning underlying resource", "resource", k8s.GetNamespacedName(krmResource))
 			return false, r.handleDeleted(ctx, krmResource)
 		}
-		if krmtotf.ShouldResolveParentForDelete(krmResource) {
-			orphaned, parent, err := r.isOrphaned(ctx, krmResource)
+		orphaned, parent, err := r.isOrphaned(ctx, krmResource)
+		// Special handling for resources that do not need the orphaned check
+		if !krmtotf.SkipOrphanedCheck(krmResource) {
```

Comment:
If all resources just deleted via status.selfLink (or equivalent), would this complex logic be needed?

Reply:
Then we won't need this logic. 
While not all resources have selfLink.  I briefly discussed this approach in my [doc](https://docs.google.com/document/d/1jLQo8clRkHw3gSs9LTMbTTM0iSzgy4fgZunXv2ka_40/edit?pli=1&resourcekey=0-C5aeWmjoceXr4JA9c1GdpQ&tab=t.0#heading=h.cne4eaxl8moi), adding a field called "relative resource name"(similar to selfLink) for all resources and use it for deletion. 
There is some concern over the scope of the fix. Introducing a new field might be "expensive", and most of existing resources do not have the problem. After investigation, only 3-4 resources have this issue on deletion. We may not want to bring a new feature in KCC at this moment. However, I agree that "relative resource name" will not only beneficial for this problem, but for some Cx requests, i.e. https://www.google.com/url?q=https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/295/&sa=D&source=docs&ust=1705093976151315&usg=AOvVaw2SGPnfgQEXAnXOLp2KvkQ9. 
If necessary, I do plan to implement this in the future, and then we can cleanup some logic.